adapter-mongodb/src/main/java/io/nosqlbench/adapter/mongodb/core/MongoOpMapper.java

```diff
@@ -39,26 +40,29 @@ public MongoOpMapper(MongodbDriverAdapter adapter) {
 
     @Override
     public OpDispenser<? extends Op> apply(ParsedOp op) {
+
         LongFunction<String> ctxNamer = op.getAsFunctionOr("space", "default");
+        String connectionValue = op.getStaticConfigOr("connection", "unknown");
+
+        if (connectionValue == null) {
+            throw new BasicError("Must provide a connection value for use be MongoDB adapter.");
+        }
+        adapter.setConnection(connectionValue);
```

Comment:
The connection should be part of the space. Adapters should be considered stateless except for the space, which is the designated state holder for each named instance.

Reply:
This makes sense.  I'll adjust those.